pykusto/retriever.py

```diff
@@ -0,0 +1,77 @@
+from concurrent.futures import Future, ThreadPoolExecutor, wait
+from itertools import chain
+from threading import Lock
+from typing import Union, Dict, Any, Iterable
+
+POOL = ThreadPoolExecutor(max_workers=1)
+
+
+class Retriever:
```

Comment:
Consider renaming to ColumnFetchers/ItemFetcher or something more indicative. And also consider renaming from items to columns (I know it's generic but it's clearer).

Reply:
Renamed to `ItemFetcher`. Calling the items columns while they are sometimes tables or databases seems even more confusing, so I kept them items.